test/langtools/tools/javac/processing/model/util/elements/TestGetDocComments.java

```diff
@@ -0,0 +1,185 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8307184
+ * @summary Test basic operation of Elements.getDocComments
+ * @library /tools/javac/lib
+ * @build   JavacTestingAbstractProcessor TestGetDocComments
+ * @compile -processor TestGetDocComments -proc:only TestGetDocComments.java
+ */
+
+import java.io.Writer;
+import java.util.*;
+import java.util.function.*;
+import javax.annotation.processing.*;
+import javax.lang.model.element.*;
+import javax.lang.model.util.*;
+
+/**
+ * Test basic workings of Elements.getDocComments
+ */
+public class TestGetDocComments extends JavacTestingAbstractProcessor {
+    public boolean process(Set<? extends TypeElement> annotations,
+                           RoundEnvironment roundEnv) {
+        if (!roundEnv.processingOver()) {
+            boolean elementSeen = false;
+
+            for (TypeElement typeRoot : ElementFilter.typesIn(roundEnv.getRootElements()) ) {
+                for (Element element : typeRoot.getEnclosedElements()) {
+                    ExpectedComment expectedComment = element.getAnnotation(ExpectedComment.class);
+                    if (expectedComment != null ) {
+                        elementSeen = true;
+                        String expectedCommentStr = expectedComment.value();
+                        String actualComment = elements.getDocComment(element);
+
+                        if (!expectedCommentStr.equals(actualComment)) {
+                            messager.printError("Unexpected doc comment found", element);
+                            System.out.println("Actual");
+                            System.out.println(actualComment);
+                            System.out.println("Expected");
+                            System.out.println(expectedCommentStr);
+                            stringDiffer(actualComment, expectedCommentStr);
+                        }
+                    }
+                }
+
+                if (!elementSeen) {
+                    throw new RuntimeException("No elements seen.");
+                }
+            }
+        }
+        return true;
+    }
+
+    void stringDiffer(String actual, String expected) {
+        if (actual.length() != expected.length()) {
+            System.out.println("Strings have different lengths");
+        }
+    }
```

Comment:
Not critical, but for bonus points, you could identify the first line that is different.

Reply:
Do you know of an existing utility usable in the JDK that does this?
(I didn't want to write a utility like that for the purpose of this bug.)

Reply:
`ToolBox.checkEqual`